lib/standard/kernel.nit

```diff
@@ -660,6 +695,120 @@ universal Byte
 	redef fun to_i is intern
 	redef fun to_f is intern
 	redef fun to_b do return self
+	redef fun to_i8 is intern
+	redef fun to_i16 is intern
+	redef fun to_u16 is intern
+	redef fun to_i32 is intern
+	redef fun to_u32 is intern
+
+	redef fun distance(i) do return (self - i).to_i
+
+	redef fun <=>(other)
+	do
+		if self < other then
+			return -1
+		else if other < self then
+			return 1
+		else
+			return 0
+		end
+	end
+
+	redef fun is_between(c, d)
+	do
+		if self < c or d < self then
+			return false
+		else
+			return true
+		end
+	end
+
+	redef fun max(other)
+	do
+		if self < other then
+			return other
+		else
+			return self
+		end
+	end
+
+	redef fun min(c)
+	do
+		if c < self then
+			return c
+		else
+			return self
+		end
+	end
+end
+
+# Native 8-bit signed integer.
+# Same as a C `int8_t`
+universal Int8
+	super Discrete
+	super Numeric
+
+	redef type OTHER: Int8
+
+	redef fun successor(i) do return self + i.to_i8
+	redef fun predecessor(i) do return self - i.to_i8
+
+	redef fun object_id is intern
+	redef fun hash do return self.to_i
+	redef fun ==(i) is intern
+	redef fun !=(i) is intern
+	redef fun output is intern
+
+	redef fun <=(i) is intern
+	redef fun <(i) is intern
+	redef fun >=(i) is intern
+	redef fun >(i) is intern
+	redef fun +(i) is intern
+
+	# On an Byte, unary minus will return `(256 - self) % 256`
+	#
+	#     assert -(1.to_i8) == 0xFF.to_i8
```

Comment:
Why not using the literal representation?

~~~nit
assert -1i8 == 1i8
assert -128i8 == -128i8 #overflow
~~~